packages/ui/src/views/repo/pull-request/components/commit-selector/commit-selector-dropdown.tsx

```diff
@@ -0,0 +1,85 @@
+import { FC, useMemo, useState } from 'react'
+
+import { DropdownMenuContent, DropdownMenuItem, Icon, SearchBox } from '@/components'
+import { cn } from '@utils/cn'
+
+import { CommitSelectorListItem } from '../../compare/components/types'
+import { CommitSelectorDropdownProps } from '../../types'
+
+const filterItems = (items: CommitSelectorListItem[], query: string) => {
+  if (!query.trim()) return items
+
+  return items.filter(item => item.title.toLowerCase().includes(query.toLowerCase().trim()))
+}
+
+export const CommitSelectorDropdown: FC<CommitSelectorDropdownProps> = ({
+  selectedCommit,
+  onSelectCommit,
+  commitList,
+
+  useTranslationStore
+}) => {
+  const [searchQuery, setSearchQuery] = useState('')
+  const { t } = useTranslationStore()
+
+  const handleSearchChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setSearchQuery(event.target.value)
```

Comment:
pls review the best practices documented here for state management: https://harness.slack.com/archives/C07QA52CWKH/p1733729112788079?thread_ts=1733525285.482889&cid=C07QA52CWKH

Reply:
done